app/javascript/src/submit_handler.js

```diff
@@ -0,0 +1,108 @@
+
+const {
+  mergeObjects
+} = require('./utilities');
+
+class SubmitHandler {
+  #config;
+  #submitEnabled
+
+  constructor($form, config) {
+    this.#config = mergeObjects({
+      text: {
+        submitted: '',
+        unsubmitted: '',
+        submitting: '',
+      },
+      buttonSelector: ':submit',
+      buttonDescriptionSelector: '',
+      preventUnload: false,
+    }, config);
+    
+    this.$form = $form;
+    this.$button = this.$form.find(this.#config.buttonSelector);
+    this.text = this.#config.text;
+    if(this.#config.buttonDescriptionSelector) {
+      this.$buttonDescription = this.$form.find(this.#config.buttonDescriptionSelector);
+    }
+
+    this.$form.on('submit', (event) => this.#handleSubmit(event));
+    this.$button.on('click', (event) => this.#handleClick(event));
+  }
+
+  get required() {
+    return this.#submitEnabled;
+  }
+
+  set required(required) {
+    if(required) { 
+      this.#enable();
+    } else {
+      this.#disable();
+    }
+  }
+
+  forceSubmit() {
+    const currentState = this.#submitEnabled;
+    this.#submitEnabled = true;
+    this.$form.submit();
+    this.#submitEnabled = currentState;
+  }
+
+  #handleClick(event) {
+    if(this.required) {
+      this.#removeBeforeUnloadListener();
```

Comment:
If a click of the button, when required is true, submits the form and the form has a submit handler that, when required is true, calls this same line, is it still needed here?